tools/lsp-wake/lsp.cpp

```diff
@@ -301,6 +301,17 @@ class LSPServer {
   static void sendMessage(const JAST &message) {
     std::stringstream str;
     str << message;
+
+    str.seekg(0, std::ios::end);
+    std::string msg = str.str();
+
+    if (msg == "{}") {
+#ifdef CERR_DEBUG
+      std::cerr << "Throwing away empty response message" << std::endl;
+#endif
+      return;
+    }
+
     str.seekg(0, std::ios::end);
```

Comment:
Yeah, it "rewinds" the string stream. Accessing the value from `std::stringstream` is analogous to playing a tape. So we need to rewind it after reading it into `std::string msg`

Reply:
Ah, didn't realize `str()` would progress the pointer.  Probably should have, but that's why I'm not (currently) working on C++.

Reply:
ah! actually out of curiosity I double checked this and `str()` doesn't advance the pointer. I'll put up a fix. 

I was using things in a different order when I started and at that point I did need the 2 rewinds

Reply:
actually this can be quite simplified :)

code has been updated

Reply:
Ah, yeah, that's much cleaner.  It doesn't break Unicode or anything?  I know C strings don't always play well with multi-byte characters.

Reply:
I think it should be fine. The main problem with unicode is that a single "character" can be multiple bytes so asking "how many bytes" is not the same as asking "how many characters" but in this case we actually do just want the number of bytes since we are sending a blob of json to be read back in on the other side